common/frame_session.go

```diff
@@ -515,9 +515,11 @@ func (fs *FrameSession) onConsoleAPICalled(event *cdpruntime.EventConsoleAPICall
 		WithTime(event.Timestamp.Time()).
 		WithField("source", "browser-console-api")
 
-	if s := fs.vu.State(); s.Group.Path != "" {
-		l = l.WithField("group", s.Group.Path)
-	}
+		/* accessing the state Group while not on the eventloop is racy
+		if s := fs.vu.State(); s.Group.Path != "" {
+			l = l.WithField("group", s.Group.Path)
+		}
+		*/
```

Comment:
I'm assuming this wouldn't be an issue if whatever action that was used was running on the event loop i.e. was async?

Reply:
yes if it was *on* the event loop it would've been fine as a `group()` call couldn't have changed the value ... as it also needs to run on the event loop.

> was async? 

async is way too generic term ... in practice it *is* async - it doesn't run syncrhonisly ;) It also doesn't run on the event loop - which is in general the idea of async ... anything. To free the event loop to do other stuff and as they don't run on the event loop (and consequently on/by the JS VM) you can do multiple stuff in parallel (or at least concurrently). The problem is that IMO "async" is used too generically. 

From what I could gather, this is just something that *always* ran asynchronously. And maybe the addition of *async* APIs made it more likely to race ... or maybe it always was possible to race - I haven't tested with a really old version of xk6-browser and `-race` :shrug: 

But yes - if we can make this code ot run on the event loop - that will make it possible to keep the lines, which is why I just commented them. Unfortunately that (IMO) will require that most if not all of the APIs are async as otherwise this code will not run if the JS VM is currently blocked on doing an *synchronous* call.

I also am not certain that this (or some of the surrounding) code does not *unblock* a certain syncrhnous call that will otherwise not continue. 

Not to speak that making this run on the event loop will be *way* bigger change, which is why I opted for dropping some info so I get rid of this race condition instead of doing something way more involved.

Reply:
I'm happy with this approach for now.